### PR TITLE
Unclaim gateway on delete

### PR DESCRIPTION
**Pull request by PavelJankoski**
#### Summary

Unclaim gateway on delete.

#### Changes

- Include unclaiming logic in the `delete-entity-header-modal` component.
##### Steps

Try deleting a managed gateway.

##### Results

The gateway should be unclaimed and then deleted.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by johanstokking — COMMENTED** [review]
LGTM, can't comment on the code, but do we need end-to-end tests too?

File: pkg/webui/console/containers/delete-entity-header-modal/index.js [hunk 1]
```diff
@@ -95,8 +95,16 @@ const deletedErrorMessageMap = {
...
-    props
+  const {
+    entity,
+    entityId,
+    entityName,
+    visible,
+    setVisible,
+    setError,
+    additionalConditions,
+    additionalAction,
```
Comment (ryaplots) [inline pkg/webui/console/containers/delete-entity-header-modal/index.js]: Now that I look at this, `additionalConditions` and `additionalAction` seems a bit vague and hard to understand.

When it comes to gateway unclaiming, I think it's better to either add a prop called `unclaimCall` and then do it if the entity is a gateway or a prop called `supports claiming` and if true unclaim the gateway.

For the additional conditions I would call the prop `isPristine`. 

```suggestion
    isPristine: isPristineProp,
    supportsClaiming,
```

File: pkg/webui/console/containers/delete-entity-header-modal/index.js [hunk 2]
```diff
@@ -138,6 +146,9 @@ const DeleteEntityHeaderModal = props => {
             if (setError) {
               setError(undefined)
             }
+            if (additionalAction) {
```
Comment (ryaplots) [inline pkg/webui/console/containers/delete-entity-header-modal/index.js]:
```suggestion
            if (supportsClaiming && isGateway) {
```

**Review by ryaplots — CHANGES_REQUESTED** [review]

**Review by ryaplots — APPROVED** [review]

**Comment by PavelJankoski** [comment]
> LGTM, can't comment on the code, but do we need end-to-end tests too?

I included e2e tests as well

**Review by johanstokking — APPROVED** [review]
Perfect, I trust the end-to-end tests.

**Review by mjamescompton — APPROVED** [review]

**Comment by KrishnaIyer** [comment]
Shall we backport this to the current `v3.32.2` release @johanstokking @PavelJankoski ?

cc: @happyRip @nicholaspcr

**Comment by johanstokking** [comment]
Yes

**Pull request merged**